### PR TITLE
[SC] Push executor functionality to VM

### PR DESCRIPTION
**Pull request by rowandh**
Pushes most executor functionality into VM create/execute methods.

### CODE REVIEW

File: src/Stratis.SmartContracts.Executor.Reflection/ReflectionVirtualMachine.cs [hunk 1]
```diff
@@ -69,57 +112,83 @@ public class ReflectionVirtualMachine : ISmartContractVirtualMachine
 
                 return VmExecutionResult.Error(gasMeter.GasConsumed, result.Exception.InnerException ?? result.Exception);
             }
-            else
-                this.logger.LogTrace("[CREATE_CONTRACT_INSTANTIATION_SUCCEEDED]");
+
+            this.logger.LogTrace("[CREATE_CONTRACT_INSTANTIATION_SUCCEEDED]");
+            
+            this.logger.LogTrace("(-):{0}={1}", nameof(contractAddress), contractAddress);
```
Comment (fassadlr) [inline src/Stratis.SmartContracts.Executor.Reflection/ReflectionVirtualMachine.cs]: This log line needs to be combined with the one below. So remove this one and include the parameters in the bottom one like so:

```
this.logger.LogTrace("(-):{0}={1}, {2}={3}", nameof(contractAddress), contractAddress, nameof(gasMeter.GasConsumed), gasMeter.GasConsumed);
```

File: src/Stratis.SmartContracts.Executor.Reflection/ReflectionVirtualMachine.cs [hunk 2]
```diff
@@ -18,48 +23,86 @@ public class ReflectionVirtualMachine : ISmartContractVirtualMachine
...
             this.logger = loggerFactory.CreateLogger(this.GetType());
+            this.network = network;
         }
 
         /// <summary>
         /// Creates a new instance of a smart contract by invoking the contract's constructor
         /// </summary>
-        public VmExecutionResult Create(byte[] contractCode,
-            ISmartContractExecutionContext context,
+        public VmExecutionResult Create(
```
Comment (fassadlr) [inline src/Stratis.SmartContracts.Executor.Reflection/ReflectionVirtualMachine.cs]: The call and create are very "heavy" now Rowam, do we intend to refactor to make them easier to read? A bit SRP there dude;)

Comment (rowandh) [reply src/Stratis.SmartContracts.Executor.Reflection/ReflectionVirtualMachine.cs]: Yes, I have more plans for these :)

**Comment by codingupastorm** [comment]
CallSmartContract and CreateSmartContract so flat! 👍

**Pull request merged**